src/web/wms/src/main/java/org/geoserver/wms/web/data/StyleAdminPanel.java

```diff
@@ -0,0 +1,498 @@
+        nameTextField.setRequired(true);
+        
+        IModel<WorkspaceInfo> wsBinding = styleModel.bind("workspace");
+        wsChoice = 
+            new DropDownChoice<WorkspaceInfo>("workspace", wsBinding, new WorkspacesModel(), new WorkspaceChoiceRenderer());
+        wsChoice.setNullValid(true);
+        if (!stylePage.isAuthenticatedAsAdmin()) {
+            wsChoice.setNullValid(false);
+            wsChoice.setRequired(true);
+        }
+
+        add(wsChoice);
+        
+        //always disable the workspace toggle if not admin
+        if (!stylePage.isAuthenticatedAsAdmin()) {
+            wsChoice.setEnabled(false);
+        }
+
+        IModel<String> formatBinding = styleModel.bind("format");
+        formatChoice = new DropDownChoice<String>("format", formatBinding, new StyleFormatsModel(), 
+                new ChoiceRenderer<String>() {
+
+            private static final long serialVersionUID = 2064887235303504013L;
+
+            @Override
+            public String getIdValue(String object, int index) {
+                return object;
+            }
+
+            @Override
+            public Object getDisplayValue(String object) {
+                return Styles.handler(object).getName();
+            }
+        });
+        formatChoice.add(new AjaxFormComponentUpdatingBehavior("change") {
+
+            private static final long serialVersionUID = -8372146231225388561L;
+
+            @Override
+            protected void onUpdate(AjaxRequestTarget target) {
+                target.appendJavaScript(String.format(
+                    "if (document.gsEditors) { document.gsEditors.editor.setOption('mode', '%s'); }", stylePage.styleHandler().getCodeMirrorEditMode()));
+            }
+        });
+        add(formatChoice);
+
+        formatReadOnlyMessage = new WebMarkupContainer("formatReadOnly", new Model<String>());
+        formatReadOnlyMessage.setVisible(false);
+        add(formatReadOnlyMessage);
+        // add the Legend fields        
+        legendPanel = new ExternalGraphicPanel("legendPanel", styleModel, stylePage.styleForm);
+        legendPanel.setOutputMarkupId(true);
+        add(legendPanel);
+        if (style.getId() != null) {
+            try {
+                stylePage.setRawStyle(stylePage.readFile(style));
+            } catch (IOException e) {
+                // ouch, the style file is gone! Register a generic error message
+                Session.get().error(new ParamResourceModel("styleNotFound", this, style.getFilename()).getString());
+            }
+        }
+        
+        // style generation functionality
+        templates = new DropDownChoice<StyleType>("templates", new Model<StyleType>(), new StyleTypeModel(), new StyleTypeChoiceRenderer());
+        templates.setOutputMarkupId(true);
+        templates.add(new AjaxFormComponentUpdatingBehavior("change") {
+
+            private static final long serialVersionUID = -6649152103570059645L;
```

Comment:
Using small anonymous serializable commands is a bother with this java format, removes some of the terse